pimd/pim_igmpv2.c

```diff
@@ -121,6 +121,13 @@ int igmp_v2_recv_report(struct igmp_sock *igmp, struct in_addr from,
 		return -1;
 	}
 
+	if (igmp_validate_checksum(igmp_msg, igmp_msg_len) == -1) {
```

Comment:
why not just call this function one time in pim_igmp_packet?

Reply:
Since the length validation is not done for each of the packet, hence not calling it. The checksum function uses igmp_msg_len which I am trying to do only after length validation for each of them.

Reply:
Also PIM_IGMP_MTRACE_RESPONSE and PIM_IGMP_MTRACE_QUERY_REQUEST packet format is different. Their checksum in the PIM header is at different location.